enable 2-erisc fabric by default on BH

SeanNijjar opened the pull request:
This changes improves fabric performance on Blackhole by work sharing across both erisc cores inside each Ethernet core. For example, neighbour exchange patterns on Line fabric double from around 16 GB/s/dir to roughly 30 GB/s/dir @ 4k packet size.

To disable, specify environment variable `TT_METAL_DISABLE_FABRIC_TWO_ERISC`

### Ticket
Closes [#20669](https://github.com/tenstorrent/tt-metal/issues/20669)

### Checklist
- [x] All post commit - T3K Quick: https://github.com/tenstorrent/tt-metal/actions/runs/19310938480
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable): https://github.com/tenstorrent/tt-metal/actions/runs/19335619871
- [x] BH Multi-card Unit Tests: https://github.com/tenstorrent/tt-metal/actions/runs/19335626348
- [x] BH Multi-card Demo Tests: https://github.com/tenstorrent/tt-metal/actions/runs/19335640166
- [x] BH Nightly: https://github.com/tenstorrent/tt-metal/actions/runs/19343214216
There is no proper BH GLX pipeline yet but I did run fabric tests, extended g08blx03 for many hours and iterations and saw no issues.

abhullar-tt reviewed (approved):
lgtm, can make the simplification Nigel suggested

SeanNijjar commented on tt_metal/fabric/erisc_datamover_builder.cpp:
```diff
@@ -72,6 +72,36 @@ namespace tt::tt_fabric {
+    bool single_erisc_dispatch = hal.get_num_risc_processors(tt::tt_metal::HalProgrammableCoreType::ACTIVE_ETH) < 2;
+
+    // 2d dynamic fabric doesn't properly support 2-erisc yet but is being deprecated anyways so we
+    // simply disable 2-erisc on it for now.
```
```suggestion
    // simply disable 2-erisc on it for now.
    // Issue [#32419](https://github.com/tenstorrent/tt-metal/issues/32419)
```

SeanNijjar committed:
fixes

SeanNijjar committed:
Update tt_metal/fabric/erisc_datamover_builder.cpp

SeanNijjar committed:
Update tt_metal/fabric/erisc_datamover_builder.cpp

SeanNijjar committed:
enable 2-erisc fabric by default on BH

To disable, specify environment variable `TT_METAL_DISABLE_FABRIC_BLACKHOLE_TWO_ERISC`

SeanNijjar committed:
forgot to skip 2-erisc fabric for dynamic routing mode
